Create Diffy_messenger_st1_ms1.md

beoblock opened the pull request:
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the Statemint account provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/pull/1560

dsm-w3f commented:
@beoblock thank you for the milestone submission. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/diffy_chat_1_dsm-w3f.md) and provide proper answers and fixes. After that, let me know to continue with the evaluation.

beoblock commented:
Hello, @dsm-w3f Diogo. Thank you for your review and sorry, that I didn't comment right away.
It seems that the issue is in sr25519 encryption and it's conversion to ed25519. Not to put these keys management on the user, we are making an appropriate fix to our Diffy code. We will update the code in a few days and I'll return it for review

semuelle commented:
Any news here, @beoblock?

beoblock commented:
Hello, @semuelle , @dsm-w3f 
Sorry for long wait.
We've upadet the code and delivery file, added new features and tests.

As transactions are signed with sr25519 keys but for encryption ed25519 is used, we have added a feature for registering a username and both keys are automatically tied to this username. P2p connection is now established using usernames. Thus users do not need to manually generate ed25519 key as encryption is all encryption manipulations are hidden inside the library. The demo video is updated as well.

dsm-w3f commented:
@beoblock thank you for the improvements and fixes. I still having some problems using the app in Firefox, in Chrome it worked. There are some problems that is still remain. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/diffy_chat_1_dsm-w3f.md) for details. Let me know when I can continue this evaluation.

StringNick commented:
@dsm-w3f we fix problem in firefox(different realisation of webrtc, support now both). Added license and fixed readme.

dsm-w3f commented:
@StringNick thank you for the improvements and fixes. I tested again and the application worked fine. Just small problems that still need to be fixed. Please see the [evaluation document](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/diffy_chat_1_dsm-w3f.md) for details. The milestone is accepted. I'll forward your invoice internally and the payment should take place within two weeks. Great job!
